pkg/apis/eventing/v1alpha1/eventtype_types.go

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2019 The Knative Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1alpha1
+
+import (
+	"github.com/knative/pkg/apis"
+	duckv1alpha1 "github.com/knative/pkg/apis/duck/v1alpha1"
+	"github.com/knative/pkg/webhook"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type EventType struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the desired state of the EventType.
+	Spec EventTypeSpec `json:"spec,omitempty"`
+
+	// Status represents the current state of the EventType.
+	// This data may be out of date.
+	// +optional
+	Status EventTypeStatus `json:"status,omitempty"`
+}
+
+// Check that EventType can be validated, can be defaulted, and has immutable fields.
+var _ apis.Validatable = (*EventType)(nil)
+var _ apis.Defaultable = (*EventType)(nil)
+var _ apis.Immutable = (*EventType)(nil)
+var _ runtime.Object = (*EventType)(nil)
+var _ webhook.GenericCRD = (*EventType)(nil)
+
+type EventTypeSpec struct {
+	Type string `json:"type,omitempty"`
```

Comment:
since this is required, leave out omitempty?

Reply:
Done! thanks!